src/openzeppelin/tests/token/erc20/test_dual20.cairo

```diff
@@ -0,0 +1,364 @@
+use traits::Into;
+use array::ArrayTrait;
+use starknet::ContractAddress;
+use starknet::contract_address_const;
+use starknet::testing::set_caller_address;
+use starknet::testing::set_contract_address;
+use openzeppelin::token::erc20::ERC20;
+use openzeppelin::token::erc20::interface::IERC20;
+use openzeppelin::token::erc20::interface::IERC20Camel;
+use openzeppelin::token::erc20::interface::IERC20Dispatcher;
+use openzeppelin::token::erc20::interface::IERC20CamelDispatcher;
+use openzeppelin::token::erc20::interface::IERC20DispatcherTrait;
+use openzeppelin::token::erc20::interface::IERC20CamelDispatcherTrait;
+use openzeppelin::token::erc20::dual20::DualERC20Trait;
+use openzeppelin::token::erc20::dual20::DualERC20;
+use openzeppelin::tests::mocks::snake20_mock::SnakeERC20Mock;
+use openzeppelin::tests::mocks::camel20_mock::CamelERC20Mock;
+use openzeppelin::tests::mocks::erc20_panic::SnakeERC20Panic;
+use openzeppelin::tests::mocks::erc20_panic::CamelERC20Panic;
+use openzeppelin::tests::mocks::non721_mock::NonERC721;
+use openzeppelin::tests::utils;
+
+use openzeppelin::tests::utils::PanicTrait;
+
+///
+/// Constants
+///
+
+const NAME: felt252 = 111;
+const SYMBOL: felt252 = 222;
+const DECIMALS: u8 = 18_u8;
+
+fn SUPPLY() -> u256 {
+    2000_u256
+}
+fn VALUE() -> u256 {
+    300_u256
+}
+fn OWNER() -> ContractAddress {
+    contract_address_const::<10>()
+}
+fn SPENDER() -> ContractAddress {
+    contract_address_const::<20>()
+}
+fn RECIPIENT() -> ContractAddress {
+    contract_address_const::<30>()
+}
+fn OPERATOR() -> ContractAddress {
+    contract_address_const::<40>()
+}
+
+///
+/// Setup
+///
+
+fn setup_snake() -> (DualERC20, IERC20Dispatcher) {
+    let mut calldata = ArrayTrait::new();
+    calldata.append(NAME);
+    calldata.append(SYMBOL);
+    calldata.append(SUPPLY().low.into());
+    calldata.append(SUPPLY().high.into());
+    calldata.append(OWNER().into());
+    let target = utils::deploy(SnakeERC20Mock::TEST_CLASS_HASH, calldata);
+    (DualERC20 { target: target }, IERC20Dispatcher { contract_address: target })
+}
+
+fn setup_camel() -> (DualERC20, IERC20CamelDispatcher) {
+    let mut calldata = ArrayTrait::new();
+    calldata.append(NAME);
+    calldata.append(SYMBOL);
+    calldata.append(SUPPLY().low.into());
+    calldata.append(SUPPLY().high.into());
+    calldata.append(OWNER().into());
+    let target = utils::deploy(CamelERC20Mock::TEST_CLASS_HASH, calldata);
+    (DualERC20 { target: target }, IERC20CamelDispatcher { contract_address: target })
```

Comment:
```suggestion
    (DualERC20 { contract_address: target }, IERC20CamelDispatcher { contract_address: target })
```

Marto changed this on [this PR](https://github.com/OpenZeppelin/cairo-contracts/blob/e68b8954a85c8c9260520a1f2ad5cc878449432d/src/openzeppelin/token/erc721/dual721.cairo#L19). Not sure with version we should settle in, but we should select one.

Reply:
Ah good call